File: data-structures/set/Set.js
function Set() {
	this.dataStore = [];
	this.add = add;
	this.remove = remove;
	this.contains = contains;
	// this.size = size;
	this.union = union;
	this.intersect = intersect;
	// this.subset = subset;
	// this.difference = difference;
	// this.show = show;
}

function add(data) {
	if (this.dataStore.indexOf(data) < 0) {
		this.dataStore.push(data);
		return true;
	} else {
		return false;
	}
}

function remove(data) {
	var position = this.dataStore.indexOf(data);
	if (position > -1) {
		this.dataStore.splice(position, 1);
		return true;
	} else {
		return false;
	}
}

function contains(data) {
	if (this.dataStore.indexOf(data) > -1) {
		return true;
	} else {
		return false;
	}
}

function union(set) {
	var tempSet = new Set();
	for (var i = 0; i < this.dataStore.length; i++) {
		tempSet.add(this.dataStore[i]);
	}
	for (var i = 0; i < set.dataStore.length; i ++) {
		if (!tempSet.contains(set.dataStore[i])) {
			tempSet.dataStore.push(set.dataStore[i]);
		}
	}
	return tempSet;
}

function intersect(set) {
	var tempSet = new Set();
	for (var i = 0; i < this.dataStore.length; i++) {
		if (set.contains(this.dataStore[i])) {
			tempSet.add(this.dataStore[i]);
		}
	}
	return tempSet;
}

File: data-structures/set/set.rb
class Set
  def initialize
  	@dataStore = []
  end

  def add(data)
    if @dataStore.index(data)
      puts "Already exists in set"
    else
      @dataStore.push(data)
      puts "#{data} added to set"
  	end
  end

  def remove(data)
  	position = @dataStore.index(data)
    @dataStore.delete_at(position)
    puts "Removed #{data} from array"
  end

  def contains(data)
    if @dataStore.index(data)
      return true
    else
      return false
    end
  end

  def union(set)
  	tempSet = Set.new;

  	@dataStore.each_with_index { |item, index| tempSet.add(@dataStore[index]) }

  	set.dataStore.each_with_index do |item, index|
  	  if !tempSet.contains(set.dataStore[index])
  	  	tempSet.dataStore.push(set.dataStore[index])
  	  end
  	end
  	return tempSet
  end

  def show
  	puts @dataStore
  end

  attr_reader :dataStore
end

# cis = Set.new
# cis.add("Clayton")
# cis.add("Mike")
# cis.add("Jennifer")
# cis.add("Raymond")

# dmp = Set.new
# dmp.add("Raymond")
# dmp.add("Cynthia")
# dmp.add("Jonathan")

# it = Set.new
# it = cis.union(dmp)
# puts it.show()
